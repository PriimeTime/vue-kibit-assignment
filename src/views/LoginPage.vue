<script setup lang="ts">
import BaseButton from "@/components/BaseButton.vue";
import BaseCard from "@/components/BaseCard.vue";
import BaseInput from "@/components/BaseInput.vue";
import HeaderText from "@/components/HeaderText.vue";
import DefaultLayout from "@/layouts/DefaultLayout.vue";
import { useRouter } from "vue-router";

const router = useRouter();
</script>

<template>
  <DefaultLayout>
    <section class="w-full flex justify-center items-center mt-10">
      <BaseCard>
        <HeaderText>Login</HeaderText>
        <form @submit.prevent class="flex flex-col">
          <BaseInput class="my-2" type="text" placeholder="Username" />
          <BaseInput class="my-2" type="password" placeholder="Password" />
          <div class="flex max-h-14 w-full">
            <BaseButton class="my-2 mr-10">Login</BaseButton>
            <BaseButton
              @click="router.push('/register')"
              flat
              class="my-2 whitespace-nowrap w-auto"
              >Register instead</BaseButton
            >
          </div>
        </form>
      </BaseCard>
    </section>
  </DefaultLayout>
</template>
