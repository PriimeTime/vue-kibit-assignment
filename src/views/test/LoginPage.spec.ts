import { describe, it, expect, vi } from "vitest";
import { mount } from "@vue/test-utils";
import LoginPage from "@/views/LoginPage.vue";
import { createTestingPinia } from "@pinia/testing";
import { createRouter, createWebHistory } from "vue-router";

// Add a mock router
const mockRouter = createRouter({
  history: createWebHistory(),
  routes: [{ path: "/", component: { template: "<div>Home</div>" } }],
});

describe("LoginPage", () => {
  it("renders without crashing", () => {
    const wrapper = mount(LoginPage, {
      global: {
        plugins: [
          mockRouter, // Provide the mock router
          createTestingPinia({
            createSpy: vi.fn,
          }),
        ],
      },
    });
    expect(wrapper.exists()).toBe(true);
  });

  it("matches snapshot", () => {
    const wrapper = mount(LoginPage, {
      global: {
        plugins: [
          mockRouter, // Provide the mock router
          createTestingPinia({
            createSpy: vi.fn,
          }),
        ],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
