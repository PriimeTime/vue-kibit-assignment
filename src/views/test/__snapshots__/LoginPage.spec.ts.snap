// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginPage > matches snapshot 1`] = `
"<header class="flex h-16 bg-teal-900">
  <div class="flex items-center w-2/3 text-xl md:text-3xl whitespace-nowrap text-white p-2 hover:cursor-pointer"> Job Ad Plaform </div>
  <div class="flex justify-end w-1/3 p-2"><button class="p-2 w-20 lg:mx-2 rounded-md text-white text-xs md:text-xl border-teal-900 border-2 bg-teal-900 flex justify-center items-center whitespace-nowrap">Home</button>
    <!--v-if--><button class="p-2 w-20 lg:mx-2 rounded-md text-white text-xs md:text-xl border-teal-900 border-2 bg-teal-900 flex justify-center items-center whitespace-nowrap">Login</button>
  </div>
</header>
<main>
  <section class="w-full flex justify-center items-center mt-10">
    <div class="max-w-[40rem] rounded-xl md:shadow-xl p-10 w-full">
      <div class="text-3xl font-bold">Login</div>
      <form class="flex flex-col">
        <div class="my-2"><input class="border-teal-900 border-2 rounded-lg p-2 w-full focus:outline-none" type="text" placeholder="Username" value=""></div>
        <!--v-if-->
        <div class="my-2"><input class="border-teal-900 border-2 rounded-lg p-2 w-full focus:outline-none" type="password" placeholder="Password" value=""></div>
        <!--v-if-->
        <div class="flex max-h-14 w-full"><button class="p-2 min-w-20 rounded-md text-white border-teal-900 border-2 bg-teal-900 hover:text-teal-900 hover:bg-white flex justify-center items-center whitespace-nowrap my-2 mr-10">Login</button><button class="p-2 min-w-20 text-teal-900 flex justify-center items-center whitespace-nowrap my-2 whitespace-nowrap w-auto">Register instead</button></div>
      </form>
    </div>
  </section>
</main>"
`;
